### PR TITLE
Function.prototype.bind: incorrect behaviour of ecma_gc_mark()

### PR DESCRIPTION
**Pull request by bzsolt**
Related issue: #531

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/base/ecma-gc.cpp [hunk 1]
```diff
@@ -352,9 +352,43 @@ ecma_gc_mark (ecma_object_t *object_p) /**< object to mark from */
             case ECMA_INTERNAL_PROPERTY_NON_INSTANTIATED_BUILT_IN_MASK_32_63: /* an integer (bit-mask) */
             case ECMA_INTERNAL_PROPERTY_REGEXP_BYTECODE:
             case ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_TARGET_FUNCTION:
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/base/ecma-gc.cpp]: Object in a [[TargetFunction]] property also need to be marked as visited.

File: jerry-core/ecma/base/ecma-gc.cpp [hunk 2]
```diff
@@ -352,9 +352,43 @@ ecma_gc_mark (ecma_object_t *object_p) /**< object to mark from */
...
+            {
+              break;
+            }
+
             case ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS:
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/base/ecma-gc.cpp]: Please, add comments near cases, related to bound functions, about what can be contained in the property.
For example, `/* an ecma-value */` for `case ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS:`.

File: jerry-core/ecma/base/ecma-gc.cpp (hunk 1 above)
Comment (bzsolt) [inline jerry-core/ecma/base/ecma-gc.cpp]: You're right, I've fixed it. Thanks.

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

**Comment by zherczeg** [comment]
LGTM. Is it possible to add a test?

**Comment by bzsolt** [comment]
I think it isn't possible, because we can't externally control the behaviour of the GC.

**Comment by egavrin** [comment]
@bzsolt I think we

**Comment by bzsolt** [comment]
@egavrin Can we merge this PR?

**Comment by ruben-ayrapetyan** [comment]
See also: #622

**Pull request merged**